pkg/autodiscovery/autoconfig.go

```diff
@@ -189,10 +165,12 @@ func (ac *AutoConfig) Stop() {
 		ac.tagFreshnessActive = false
 	}
 
-	// stop the config poller if running
-	if ac.pollerActive {
-		ac.pollerStop <- struct{}{}
-		ac.pollerActive = false
+	// stop polled config providers
+	for _, pd := range ac.providers {
+		if !pd.isPolling {
```

Comment:
I think this check is not necessary, we should always stop the provider and do the check inside the function